modules/tensor_mechanics/src/materials/crystal_plasticity/CrystalPlasticityKalidindiUpdate.C

```diff
@@ -108,81 +91,59 @@ CrystalPlasticityKalidindiUpdate::calculateConstitutiveEquivalentSlipIncrement(
       mooseWarning("Maximum allowable slip increment exceeded ",
                    std::abs(_slip_increment[_qp][i]) * _substep_dt);
 #endif
-      error_tolerance = true;
-      return;
+      return false;
     }
   }
-
-  // Sum up the slip increments to find the equivalent plastic strain due to slip
-  for (unsigned int i = 0; i < _number_slip_systems; ++i)
-    equivalent_slip_increment += _flow_direction[_qp][i] * _slip_increment[_qp][i] * _substep_dt;
-
-  error_tolerance = false;
+  return true;
 }
 
 void
 CrystalPlasticityKalidindiUpdate::calculateConstitutiveSlipDerivative(
-    std::vector<Real> & dslip_dtau, unsigned int /*slip_model_number*/)
+    std::vector<Real> & dslip_dtau)
 {
   for (unsigned int i = 0; i < _number_slip_systems; ++i)
   {
     if (MooseUtils::absoluteFuzzyEqual(_tau[_qp][i], 0.0))
```

Comment:
why this check? shouldn't the expression in the `else` clause also return 0 in this case?

Reply:
I guess because when `_tau[_qp][i] = 0` the evaluation will be faster by not carrying out the calculation in the `else` clause?